diff --git a/node_modules/hexo-filter-fenced-code/lib/filter.js b/node_modules/hexo-filter-fenced-code/lib/filter.js
index 10c0dac..28951fb 100644
--- a/node_modules/hexo-filter-fenced-code/lib/filter.js
+++ b/node_modules/hexo-filter-fenced-code/lib/filter.js
@@ -33,7 +33,7 @@
  *   `````
  * */
 
-var rFenceCode = /(\s*)(`{3,}|~{3,}) *(.*) *\n?([\s\S]+?)\s*(\2)(\n+|$)/g;
+var rFenceCode = /(\s*)(`{3,}|~{3,}) *(.*)\s*\n?([\s\S]+?)\s*(\2)(\n+|$)/g;
 var rLang = /([^\s]+)\s*(.+)?\s*(.+)?/;
 
 var langMap = {
@@ -82,7 +82,8 @@ var blockTags = [
   'codeblock',
   'pullquote',
   'div',
-  'section'
+  'section',
+  'mermaid'
 ];
 
 var inlineTags = [
